admin/class-admin-apple-news.php

```diff
@@ -135,13 +135,21 @@ public function __construct() {
 			add_action(
 				'rest_api_init',
 				function() {
-					register_rest_field(
-						'post',
-						'apple_news_notices',
-						[
-							'get_callback' => [ 'Admin_Apple_Notice', 'get_if_allowed' ],
-						]
-					);
+					$admin_settings = new Admin_Apple_Settings();
+					$post_types     = $admin_settings->fetch_settings()->post_types ?? [];
+					if ( ! is_array( $post_types ) ) {
```

Comment:
I don't think that the `post_types` key would ever be anything other than an array, so I don't believe this part should be necessary. There are other places in the code where we're using the value directly without doing an array check first.